File: one_fm/one_fm/report/7_day_consecutive_absences/7_day_consecutive_absences.py
from frappe import _

from one_fm.utils import get_consecutive_absences_data

def execute(filters=None):
    columns = get_columns()
    data = get_consecutive_absences_data(filters)
    return columns, data

def get_columns():
    return [
        _("Employee") + ":Link/Employee:250",
        _("Employee Name") + ":Data:250",
        _("Department") + ":Data:250",
        _("Start Date") + ":Date:150",
        _("End Date") + ":Date:150",
        _("No of Days") + ":Data:150",
    ]

File: one_fm/one_fm/report/7_day_consecutive_absences/7_day_consecutive_absences.js
// Copyright (c) 2025, omar jaber and contributors
// For license information, please see license.txt


frappe.query_reports["7-Day Consecutive Absences"] = {
	"filters": [
			{
			fieldname: "employee",
			label: __("Employee"),
			fieldtype: "Link",
			options: "Employee",
		},
			{
			fieldname: "department",
			label: __("Department"),
			fieldtype: "Link",
			options: "Department",
		},
		{
			fieldname: "start_date",
			label: __("Start Date"),
			fieldtype: "Date",
			reqd: 1,
			default: frappe.datetime.add_days(frappe.datetime.get_today(), -6)
		},
		{
			fieldname: "end_date",
			label: __("End Date"),
			fieldtype: "Date",
			reqd: 1,
			default: frappe.datetime.get_today()
		}
	]
};